File: Java/09-Laço Do While/DoWhile.java
import java.util.Scanner;

class DoWhile {
    public static void main(String args[]) {
        // O laço do while executa o bloco pelo menos uma vez pois o teste só é feito no
        // final do bloco
        Scanner entrada = new Scanner(System.in);
        int x;
        int y;
        String sair;

        do {
            System.out.println("Calculadora de subtracao");
            System.out.println("Digite o primeiro valor :");
            x = entrada.nextInt();

            System.out.println("Digite o segundo valor :");
            y = entrada.nextInt();
            entrada.nextLine();// Necessario para usar scanner em sair

            int sub = x - y;

            System.out.println(x + " - " + y + " = " + sub);
            System.out.println("Deseja sair s / n");
            sair = entrada.nextLine();

        } while (!(sair.equals("s")));

        // Enquanto sair NÂO receber s este laço será executado
        entrada.close();
    }
}

File: PHP/09-Laço Do While/DoWhile.php
<?php
    
    //No laço do/while o teste da condição de repetição só e feito no final
    //isto garante a execução do bloco pelo menos 1 vez independente do
    //resultado do teste

    $contador = 0;
    
    do {

        echo $contador . "<br/>";
        $contador++;
        
    } while ($contador <= 10);

?>